### PR TITLE
PMM-10218 Broken Pprof file returned on concurrent pprof request.

### PR DESCRIPTION
**Pull request by pkadej**
[PMM-10218](https://jira.percona.com/browse/PMM-10218) Broken Pprof file returned on concurrent pprof request

Build: SUBMODULES-2661

### CODE REVIEW

**Comment by ritbl** [comment]
@pkadej you can lock resource, in case it is locked you can return  something like `429` (Too Many Requests) or other 4xx.

**Review by ritbl — APPROVED** [review]
LGTM

File: managed/services/supervisord/logs.go [hunk 1]
```diff
@@ -217,38 +217,45 @@ func (l *Logs) files(ctx context.Context, pprofConfig *PprofConfig) []fileConten
 
 	// add pprof
 	if pprofConfig != nil {
+		filesSync := &sync.Mutex{}
```
Comment (pkadej) [inline managed/services/supervisord/logs.go:220]: This one is not related to PR's goal. 
I found race problem (with appending to files array) and decided to fix it.

Comment (artemgavrilov) [inline managed/services/supervisord/logs.go:220]:
```suggestion
		var filesSync sync.Mutex
```

**Review by artemgavrilov — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]
